Bang/GameEvents/CharacterEffects/States/WaitingForCardSelectionState.cs

```diff
@@ -13,6 +13,12 @@ public WaitingForCardSelectionState(Gameplay gameplay) : base(gameplay)
 
         public override HandlerState ApplyCardEffect(Player player, BangGameCard card)
         {
+            if (!((ChooseCardsResponse)SideEffect).CardsToChoose.Contains(card))
```

Comment:
It would be more OOP-way if we store `cardsToChoose` as private field and inject it in constructor